src/delete.rs

```diff
@@ -0,0 +1,177 @@
+    let mut cli: Vec<String> = vec![];
+
+    if docker {
+        let user_home = home_dir().expect("unable to retrieve home directory");
+        let kube_config =
+            env::var("KUBECONFIG").unwrap_or(format!("{}/.kube/config", user_home.display()));
+        cli.extend(
+            [
+                "docker",
+                "run",
+                "--interactive",
+                "--rm",
+                "--network",
+                "host",
+                "-v",
+                &format!("{}:/.kube/config", kube_config),
+                "--env",
+                "KUBECONFIG=/.kube/config",
+                KUBECTL_IMAGE,
+            ]
+            .iter()
+            .map(|s| s.to_string())
+            .collect::<Vec<_>>(),
+        );
+    } else {
+        cli.extend(
+            ["kubectl"]
+                .iter()
+                .map(|s| s.to_string())
+                .collect::<Vec<_>>(),
+        );
+    }
+
+    cli.extend(
+        [
+            "delete",
+            "configmap",
+            &cleanup_hack_resource_name(app_instance),
+            "--namespace",
+            &app_instance.namespace_any(),
+        ]
+        .iter()
+        .map(|s| s.to_string())
+        .collect::<Vec<_>>(),
+    );
+
+    cli
+}
+
+/// Create a blank ConfigMap. This is used as a utility to help cleanup
+/// resources by leveraging the applyset functionality.
+///
+/// Unfortunately, we cannot use a blank object of kind `List` as the applyset
+/// requires that _some_ objects are passed to it.
+pub fn emit_deletion_setup(name: &str, namespace: &str) -> Vec<String> {
+    [
+        "kubectl",
+        "create",
+        "configmap",
+        name,
+        "--namespace",
+        namespace,
+        "--dry-run=client",
+        "-o=yaml",
+    ]
+    .iter()
+    .map(|s| s.to_string())
+    .collect::<Vec<_>>()
+}
+
+/// Utility to generate the cleanup configmap name based on a given name.
+pub fn cleanup_hack_resource_name(app_instance: &AppInstance) -> String {
+    format!("{}-cleanup", app_instance.name_any())
+}
+
+/// Generates a shell script that will cleanup the created AppInstance resources.
+pub fn script(app_instance: &AppInstance, deletion_dir: &str, docker: bool) -> Result<Script> {
+    let tokens = emit_commandline(app_instance, deletion_dir, docker);
+    Ok(Script::from_vec(tokens))
+}
+
+/// Generates a shell script that is used post prune operation of the AppInstance
+/// resources. In other words, it is used to delete the blank ConfigMap that was
+/// used as the blank applyset.
+pub fn post_pruning_script(app_instance: &AppInstance, docker: bool) -> Result<Script> {
+    let configmap_deletion = emit_post_deletion_commandline(app_instance, docker);
+    Ok(Script::from_vec(configmap_deletion))
+}
+
+/// Generates a shell script that is used as a helper during the cleanup process
+/// of the associated AppInstance.
+pub fn setup_script(app_instance: &AppInstance) -> Result<Script> {
```

Comment:
This is also missing the `docker: bool` portion :facepalm: Otherwise it'll mean that it _always_ runs with simply `kubectl` which might not be the right version when doing `local delete ...` .

For completeness I'll add this now.

Reply:
also running `kubit` itself from the script spawned by kubit may not work if somebody doesn't add the kubit executable to the PATH. (or worse, when somebody has the prebuilt version in the path, but they are running the indevelopment version with cargo run!)